tests/data/test_dataset_registry.py

```diff
@@ -0,0 +1,133 @@
+from typing import Optional
+
+import pytest
+from unittest import mock
+
+import mlflow.data
+from mlflow.data.dataset import Dataset
+from mlflow.data.dataset_registry import DatasetRegistry, register_constructor
+from mlflow.data.dataset_source_registry import resolve_dataset_source, DatasetSourceRegistry
+from mlflow.exceptions import MlflowException
+
+from tests.resources.data.dataset import TestDataset
+from tests.resources.data.dataset_source import TestDatasetSource
+
+
+@pytest.fixture
+def dataset_source_registry():
+    registry = DatasetSourceRegistry()
+    with mock.patch("mlflow.data.dataset_source_registry._dataset_source_registry", wraps=registry):
+        yield registry
+
+
+@pytest.fixture
+def dataset_registry():
+    registry = DatasetRegistry()
+    with mock.patch("mlflow.data.dataset_registry._dataset_registry", wraps=registry):
+        yield registry
+
+
+def test_register_constructor_function_performs_validation():
+    registry = DatasetRegistry()
+
+    def from_good_function(
+        path: str, name: Optional[str] = None, digest: Optional[str] = None
+    ) -> Dataset:
+        pass
+
+    registry.register_constructor(from_good_function)
+
+    def bad_name_fn(name: Optional[str] = None, digest: Optional[str] = None) -> Dataset:
+        pass
+
+    with pytest.raises(MlflowException, match="Constructor name must start with"):
+        registry.register_constructor(bad_name_fn)
+
+    with pytest.raises(MlflowException, match="Constructor name must start with"):
+        registry.register_constructor(
+            constructor_fn=from_good_function, constructor_name="bad_name"
+        )
+
+    def from_no_name_fn(digest: Optional[str] = None) -> Dataset:
+        pass
+
+    with pytest.raises(MlflowException, match="must define an optional parameter named 'name'"):
```

Comment:
Nit: this message feels weird...where is it coming from?

Reply:
It's not a user-facing error, it's a developer-facing error when they attempt to register a bad dataset constructor function.